### DISCUSSION TITLE
"Upload failed" while importing a 20MB Confluence export file

### DISCUSSION THREAD

**Discussion by ultimatebash** [outline/outline#7333] [Self Hosting]
We would like to migrate from Confluence Cloud to Outline, so we are looking at the self-hosted edition of Outline through a trial license.

First we tried it with smaller confluence exports, about 50KB in size, which could be imported without any problems.

When we tried to import a larger space (around 20MB) today, we got the error message “Upload failed”

in the development console of the Chrome browser it says:
`POST https://outline.xyz.xyz.lan/api/files.create 413 (Payload Too Large)`

the belonging log file of the container: 
`{"label":"http","level":"info","message":"  <-- GET /settings/import"}
{"label":"http","level":"info","message":"  --> GET /settings/import 200 44ms 1.56kb"}
{"label":"http","level":"info","message":"  <-- POST /api/auth.info"}
{"label":"http","level":"info","message":"  --> POST /api/auth.info 200 48ms 2.05kb"}
{"label":"worker","level":"info","message":"ValidateSSOAccessTask running","userId":"e726c245-564c-4fd2-9d12-2af097c9dd69"}
{"label":"authentication","level":"debug","message":"No refresh token found, skipping refresh"}
{"label":"http","level":"info","message":"  <-- POST /api/notifications.list"}
{"label":"http","level":"info","message":"  --> POST /api/notifications.list 200 36ms 166b"}
{"label":"websockets","level":"debug","message":"Authenticated socket CXiQagmYFzMuorFpAAAH"}
{"label":"http","level":"info","message":"  <-- POST /api/documents.list"}
{"label":"http","level":"info","message":"  <-- POST /api/fileOperations.list"}
{"label":"http","level":"info","message":"  --> POST /api/documents.list 200 26ms 147b"}
{"label":"http","level":"info","message":"  --> POST /api/fileOperations.list 200 13ms 138b"}
{"label":"http","level":"info","message":"  <-- GET /static/sw.js"}
{"label":"http","level":"info","message":"  --> GET /static/sw.js 200 6ms 38.13kb"}
{"label":"http","level":"info","message":"  <-- POST /api/notifications.list"}
{"label":"http","level":"info","message":"  <-- POST /api/documents.drafts"}
{"label":"http","level":"info","message":"  <-- POST /api/collections.list"}
{"label":"http","level":"info","message":"  --> POST /api/documents.drafts 200 29ms 139b"}
{"label":"http","level":"info","message":"  --> POST /api/notifications.list 200 42ms 166b"}
{"label":"http","level":"info","message":"  <-- POST /api/userMemberships.list"}
{"label":"http","level":"info","message":"  <-- POST /api/stars.list"}
{"label":"http","level":"info","message":"  --> POST /api/stars.list 200 21ms 158b"}
{"label":"http","level":"info","message":"  --> POST /api/collections.list 200 41ms 795b"}
{"label":"http","level":"info","message":"  --> POST /api/userMemberships.list 200 33ms 174b"}
{"label":"http","level":"info","message":"  <-- POST /api/documents.viewed"}
{"label":"http","level":"info","message":"  <-- POST /api/pins.list"}
{"label":"http","level":"info","message":"  --> POST /api/pins.list 200 29ms 156b"}
{"label":"http","level":"info","message":"  --> POST /api/documents.viewed 200 58ms 4.69kb"}
{"label":"http","level":"info","message":"  <-- POST /api/collections.documents"}
{"label":"http","level":"info","message":"  --> POST /api/collections.documents 200 18ms 531b"}
{"label":"http","level":"info","message":"  <-- POST /api/notifications.list"}
{"label":"http","level":"info","message":"  --> POST /api/notifications.list 200 15ms 166b"}
{"label":"http","level":"info","message":"  <-- POST /api/fileOperations.list"}
{"label":"http","level":"info","message":"  --> POST /api/fileOperations.list 200 10ms 138b"}
{"label":"http","level":"info","message":"  <-- POST /api/attachments.create"}
{"label":"http","level":"info","message":"  --> POST /api/attachments.create 200 26ms 597b"}
{"attempt":0,"event":{"actorId":"e726c245-564c-4fd2-9d12-2af097c9dd69","changes":null,"collectionId":null,"createdAt":"2024-07-31T15:05:02.255Z","data":{"name":"Confluence-space-export-142350.html.zip"},"documentId":null,"id":"1fb3afb8-a59b-40c9-a3cd-de36a071ec2d","ip":"0.0.0.0","modelId":"c68aab37-82ce-45b6-83bc-d10a22c6f99e","name":"attachments.create","teamId":"4c136d44-92a5-4e04-a858-5617152e2b11","userId":null},"label":"worker","level":"info","message":"Processing attachments.create"}
{"event":{"actorId":"e726c245-564c-4fd2-9d12-2af097c9dd69","changes":null,"collectionId":null,"createdAt":"2024-07-31T15:05:02.255Z","data":{"name":"Confluence-space-export-142350.html.zip"},"documentId":null,"id":"1fb3afb8-a59b-40c9-a3cd-de36a071ec2d","ip":"0.0.0.0","modelId":"c68aab37-82ce-45b6-83bc-d10a22c6f99e","name":"attachments.create","teamId":"4c136d44-92a5-4e04-a858-5617152e2b11","userId":null},"label":"worker","level":"info","message":"GleanProcessor running attachments.create"}
{"event":{"actorId":"e726c245-564c-4fd2-9d12-2af097c9dd69","changes":null,"collectionId":null,"createdAt":"2024-07-31T15:05:02.255Z","data":{"name":"Confluence-space-export-142350.html.zip"},"documentId":null,"id":"1fb3afb8-a59b-40c9-a3cd-de36a071ec2d","ip":"0.0.0.0","modelId":"c68aab37-82ce-45b6-83bc-d10a22c6f99e","name":"attachments.create","teamId":"4c136d44-92a5-4e04-a858-5617152e2b11","userId":null},"label":"worker","level":"info","message":"WebhookProcessor running attachments.create"}
`
ip-addresses are obscured

The storage, we are using is local, no cloud storage is used

The 4 environment variables from the docker install guide (File storage) are set 

picture of the files.ts
![file](https://github.com/user-attachments/assets/61300418-60d8-467a-9e02-3770370ad012)

**Comment by tommoor** [comment]
Please see the various configuration variables available here, in particular `FILE_STORAGE_WORKSPACE_IMPORT_MAX_SIZE`

https://docs.getoutline.com/s/hosting/doc/file-storage-N4M0T6Ypu7#h-environment-variables

**Comment by ultimatebash** [comment]
our fault, it was a proxy issue
